### PR TITLE
Add support for viewing trashed posts

### LINKED ISSUE #6037: Posts: Allow trashed posts to be previewed.
Currently, if you tap on a trashed post in the post list you can see the tap animation (probably a minor bug tbh), but you can't view the post's content. This is consistent with the web experience, but it might be nice if a user could view the full post content for extra context before restoring a trashed post.

### PR DESCRIPTION
**Pull request by kean**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/6037 by allowing you to preview trashed posts and restore them right from the editor. This is now a much better experience than the production version where you tap on a trashed post and nothing happens. On Android, tapping on a trashed prompt, opens an alert to restore it. This is taking it a step further.

## Testing

**Test 1.1**

**Note**: test for Posts, Pages and Search

- Open a trashed post
- ✅ Verify that navigation bar buttons are hidden
- Tap anywhere in the UI or perform any other gesture like swip
- ✅ Verify that the alert promting you to restore the post is shown
- Tap "Restore"
- ✅ Verify that the success notice was shown
- ✅ Verify that the UI was reloaded and the navigation bar buttons appeared (matching "draft" state)
- (Optional) Make and save changes to the post
- ✅ Verify that the changes were saved

**Test 1.2**

- Open a trashed post
- Close the trashed post
- Open the trashed post
- ✅ Verify that no assertions were hit and nothing updated in the UI

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/9f66ee3a-c114-469b-9f49-088b4b26ccfe

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/PostEditor.swift [hunk 1]
```diff
@@ -242,4 +299,10 @@ private enum Strings {
 
     static let autosaveAlertContinue = NSLocalizedString("autosaveAlert.viewChanges", value: "View Changes", comment: "An alert suggesting to load autosaved revision for a published post")
     static let autosaveAlertCancel = NSLocalizedString("autosaveAlert.cancel", value: "Cancel", comment: "An alert suggesting to load autosaved revision for a published post")
+
+    static let trashedPostSheetTitle = NSLocalizedString("postEditor.recoverTrashedPostAlert.title", value: "Trashed %@", comment: "Editor, alert for recovering a trashed post")
+    static let trashedPostSheetMessage = NSLocalizedString("postEditor.recoverTrashedPostAlert.message", value: "Trashed %1$@ can't be edited. To edit this %1$@, you'll need to restore it by moving it back to a draft.", comment: "Editor, alert for recovering a trashed post")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Post/PostEditor.swift]:
```suggestion
    static let trashedPostSheetMessage = NSLocalizedString("postEditor.recoverTrashedPostAlert.message", value: "A trashed %1$@ can't be edited. To edit this %1$@, you'll need to restore it by moving it back to a draft.", comment: "Editor, alert for recovering a trashed post")
```

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -432,6 +432,8 @@
 		0A9687BC28B40771009DCD2F /* FullScreenCommentReplyViewModelMock.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0A9687BB28B40771009DCD2F /* FullScreenCommentReplyViewModelMock.swift */; };
 		0C01A6EA2AB37F0F009F7145 /* SiteMediaCollectionCellSelectionOverlayView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0C01A6E92AB37F0F009F7145 /* SiteMediaCollectionCellSelectionOverlayView.swift */; };
 		0C01A6EB2AB37F0F009F7145 /* SiteMediaCollectionCellSelectionOverlayView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0C01A6E92AB37F0F009F7145 /* SiteMediaCollectionCellSelectionOverlayView.swift */; };
+		0C02E6C52BE3B6E30055F0F6 /* PostTrashedOverlayView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0C02E6C42BE3B6E30055F0F6 /* PostTrashedOverlayView.swift */; };
```
Comment (momo-ozawa) [inline WordPress/WordPress.xcodeproj/project.pbxproj:435]: [nit] rename to TrashedPostOverlayView.swift to match class name

**Review by momo-ozawa — APPROVED** [review]
LGTM with a few nits

File: WordPress/WordPress.xcodeproj/project.pbxproj (hunk 2 above)
Comment (kean) [reply WordPress/WordPress.xcodeproj/project.pbxproj:435]: 👍  I renamed the class itself to PostTrashedOverlayView – using `Post` as a namescape.

**Pull request merged**